docs/documentation/server_admin/topics/workflows/understanding-workflows-engine.adoc

```diff
@@ -0,0 +1,61 @@
+[id="understanding-workflow_{context}"]
+
+[[_understanding_workflows_engine_]]
+=  Understanding the workflows engine
+[role="_abstract"]
+
+The lifecycle and execution of workflows in {project_name} is managed by the Workflows Engine.
+The engine is responsible for processing events, creating workflow executions, and processing their steps.
+Once a workflow execution is created, the engine takes over and manages the execution of its steps chain.
+
+The execution of the steps chain is done asynchronously and detached from the request that originated the event that
+triggered the workflow. This means that workflows are not scheduled nor executed immediately within the context of the request,
+but rather sent to a task thread pool and scheduled to be executed later by the engine. This allows for better performance and scalability, as the steps can be executed
+in the background without blocking the main request thread.
+
+For immediate steps, the engine processes them as soon as the workflow execution is submitted to the task thread pool. This means that immediate steps are executed
+right away, as soon as the workflow execution is picked up by a worker thread from the pool.
+
+For scheduled steps, the engine continuously monitors the state of all active workflow executions in the realm. This is done by a background
+task that runs periodically to check for any workflow executions that have steps ready to be executed. The schedule steps of a workflow are also executed asynchronously
+by the background task, using the same task thread pool as immediate steps.
```

Comment:
Right now an administrator has to give some thought to figuring out what is an immediate step and what will be a scheduled step.  It might help readability to have some explicit way of letting the reader of a YAML know which is which.

Reply:
@pedroigor I agree.   The former lgtm.